### PR TITLE
DataScience: Visualization and AzureStorageDownloader major improvements

### PR DESCRIPTION
**Pull request by marco-rossi29**
1) Decoupled logDownloader logic from Visualization.py. logD logic is now part of AzureStorageDownloader.py (version=1 flag)
2) Refactored cmd line parsing to avoid code duplication
3) Enabled reading Account name and key from ds.config (when pickle file doesn't exist)
4) ds.config: Removed unused settings and moved needed ones to cmd line input
5) Check that AzureStorage account_key entered has length 88
6) Using get_blob_to_path() to prevent saturating memory when downloading large files
7) Changed legends labels for Online performance and Baseline estimate
8) Cosmetic improvements

### CODE REVIEW

**Comment by marco-rossi29** [comment]
This PR is mainly for Alekh. Its priority is low otherwise.

File: DataScience/AzureStorageDownloader.py [hunk 1]
```diff
@@ -1,127 +1,176 @@
...
+                diff_time = curr_time-bp.properties.last_modified
+                will_download = (not skip_current) or diff_time > datetime.timedelta(0, 600, 0)    # download blob if not modified in the last 10 min
+                if will_download:
+                    print("\nBlob {}\nUncompressed Size: {:.3f} MB\nLast modified: {}\nCurrent time: {}\nDiff from current time: {} - Will download: {}".format(blob.name, bp.properties.content_length/(1024**2), bp.properties.last_modified, curr_time, diff_time, will_download))
+                    print('Destination: {}\nDownloading...'.format(fp), end='')
+                    if dry_run:
+                        print(' dry_run!')
+                    else:
+                        t0 = time.time()
+                        bbs.get_blob_to_path(container, blob.name, fp)
```
Comment (eisber) [inline DataScience/AzureStorageDownloader.py]: do you already get intermediate progress? if not this method provides a progress_callback invoked regularly to give user feedback

File: DataScience/AzureStorageDownloader.py [hunk 2]
```diff
@@ -1,127 +1,176 @@
...
-                    start_date = datetime.datetime.strptime(sys.argv[ii], '%Y-%m-%d')
-                elif sys.argv[ii] == '--end_date':
-                    ii += 1
-                    end_date = datetime.datetime.strptime(sys.argv[ii], '%Y-%m-%d')
-                else:
-                    print('Input arg: {} not recognized! Abort!'.format(sys.argv[ii]))
-                    print(usage_str)
-                    sys.exit()
+        ii = 1
+        while ii < len(argv):
```
Comment (eisber) [inline DataScience/AzureStorageDownloader.py]: any reason not to use lib/argparse or lib/optparse?

File: DataScience/ds.config [hunk 3]
```diff
@@ -10,10 +10,3 @@ ExperimentalUnitDurationDays: 1
 [Experimentation]
 LearningRates: 0.5,0.25,0.1,0.05,0.025,0.01,0.005,0.0025,0.001,0.0005,0.00025,0.0001,0.00005,0.000025,0.00001
 RegularizationValues: 0,1e-1,1e-2,1e-3,1e-4,1e-5,1e-6,1e-7,1e-8,1e-9
-
-[Visualization]
-LogDownloaderURL: https://cps-staging-exp-experimentation.azurewebsites.net/api/Log?account={ACCOUNT}&key={KEY}&start={START}&end={END}&container={CONTAINER}
```
Comment (eisber) [inline DataScience/ds.config]: why is this removed?

Comment (marco-rossi29) [reply DataScience/ds.config]: It's moved to AzureStorageDownloader.py

After a discussion with Alekh, he suggested avoiding having it in ds.config to. Since this is only used on AzureStorageDownloader.py, I moved it there. I don't have a strong feeling about this, so we can go either way with this

File: DataScience/AzureStorageDownloader.py (hunk 1 above)
Comment (marco-rossi29) [reply DataScience/AzureStorageDownloader.py]: I didn't implement intermediate progress. I will look into it

File: DataScience/AzureStorageDownloader.py (hunk 2 above)
Comment (marco-rossi29) [reply DataScience/AzureStorageDownloader.py]: no reason. I can do argparse
